### PR TITLE
INFRA-3543: prune AWS regions + remove dockerhub

### PR DESCRIPTION
**Pull request by aastein**
- Create an ECR Client for each region in the regions list
- Delete images for each ECR Client

Tested locally with DRY_RUN=true

refs: https://github.com/Clever/ark-config/pull/1952

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -2,65 +2,100 @@ package main
...
+
+		awsConfig := aws.NewConfig().WithMaxRetries(10)
+		awsConfig.Region = &region
+		ecrClient := ecr.New(session.New(), awsConfig)
 
-	for _, repo := range repos {
-		kv.DebugD("dockerhub-get-repo-tags", logger.M{"repo": repo})
-		repoTags, err := dockerhubClient.GetTagsForRepo(repo)
+		// Get repositories in region
+		repositories, err := ecrClient.DescribeRepositories(&ecr.DescribeRepositoriesInput{})
```
Comment (rgarcia) [inline handler.go]: probably need to paginate here via `DescribeRepositoriesPages`

File: handler.go [hunk 2]
```diff
@@ -2,65 +2,100 @@ package main
...
-		}
+	return nil
+}
 
-		// Prune ECR with same image tags that were pruned from Docker Hub
-		kv.InfoD("ecr-prune-repo", logger.M{"repo": repo})
-		ecrClient.DeleteImages(deletedImages)
+func pruneRepo(ecrClient *ecr.ECR, repo *ecr.Repository) error {
+	kv.DebugD("ecr-get-repo-images", logger.M{"repo": *repo.RepositoryName})
+	images, err := ecrClient.DescribeImages(&ecr.DescribeImagesInput{
```
Comment (rgarcia) [inline handler.go]: same here, `DescribeImagesPages`

File: handler.go [hunk 3]
```diff
@@ -2,65 +2,100 @@ package main
...
+	if len(images.ImageDetails) <= config.MinImagesInRepo {
+		return nil
+	}
+
+	kv.InfoD("ecr-prune-repo", logger.M{"repo": *repo.RepositoryName, "image_count": len(images.ImageDetails)})
+
+	// Sort the images from most recent to least recent
+	sort.Slice(images.ImageDetails, func(i, j int) bool {
+		return images.ImageDetails[i].ImagePushedAt.Unix() > images.ImageDetails[j].ImagePushedAt.Unix()
+	})
```
Comment (rgarcia) [inline handler.go]: 👌

File: handler.go [hunk 4]
```diff
@@ -2,65 +2,115 @@ package main
...
+
+	imagesToDelete := []*ecr.ImageIdentifier{}
+
+	for i := config.MinImagesInRepo; i < len(images); i++ {
+		image := images[i]
+		kv.DebugD("image-to-delete", logger.M{"repo": *repo.RepositoryName, "registry": *repo.RegistryId, "image": *image.ImageDigest})
+		imagesToDelete = append(imagesToDelete, &ecr.ImageIdentifier{
+			ImageDigest: image.ImageDigest,
+		})
```
Comment (rgarcia) [inline handler.go]: I'm wondering if we should have a sanity check here, e.g. don't delete any images created in the last week

Comment (aastein) [reply handler.go]: That seems like a good idea

**Pull request merged**